src/main/java/net/minecraftforge/registries/DataPackRegistriesHooks.java

```diff
@@ -0,0 +1,101 @@
+package net.minecraftforge.registries;
+
+import com.google.common.collect.ImmutableMap;
+import net.minecraft.core.Holder;
+import net.minecraft.core.Registry;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.data.BuiltinRegistries;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraftforge.event.world.AddBuiltinRegistryEvent;
+import net.minecraftforge.fml.ModLoader;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class DataPackRegistriesHooks
+{
+   private static Map<ResourceKey<? extends Registry<?>>, RegistryAccess.RegistryData<?>> REGISTRY_ACCESS_REGISTRIES_COPY;
+
+   /** Modders, DO NOT USE. Internal use only */
+   public static Map<ResourceKey<? extends Registry<?>>, RegistryAccess.RegistryData<?>> addBuiltinRegistriesEvent(ImmutableMap.Builder<ResourceKey<? extends Registry<?>>, RegistryAccess.RegistryData<?>> builder)
+   {
+      REGISTRY_ACCESS_REGISTRIES_COPY = new HashMap<>(builder.build());
+      return REGISTRY_ACCESS_REGISTRIES_COPY;
+   }
+
+   /** Modders, DO NOT USE. Internal use only */
+   public static void fireAddBuiltinRegistriesEvent()
+   {
+      AddBuiltinRegistryEvent registryEvent = new AddBuiltinRegistryEvent((resourceKey, registryData) ->
+      {
+         validateRegistryKey(resourceKey);
+         if (!BuiltinRegistries.REGISTRY.containsKey(resourceKey.location()))
+         {
+            throw new IllegalArgumentException("DataPack registry must be marked as DataPack Registry in RegistryBuilder");
```

Comment:
Weird formatting, this should probably directly reference the method on RegistryBuilder in the format `RegistryBuilder#isDataPackRegistry` and DataPack capitalization is weird.